test/common/tcp_proxy/upstream_test.cc

```diff
@@ -183,6 +183,17 @@ TYPED_TEST(HttpUpstreamTest, OnFailureCalledOnInvalidResponse) {
   this->upstream_->responseDecoder().decodeHeaders(std::move(headers), false);
 }
 
+TYPED_TEST(HttpUpstreamTest, PropagateResponseHeadersInCaseOfInvalidResponse) {
+  setupUpstream();
+  auto conn_pool_callbacks = std::make_unique<MockHttpConnPoolCallbacks>();
+  auto conn_pool_callbacks_raw = conn_pool_callbacks.get();
+  upstream_->setConnPoolCallbacks(std::move(conn_pool_callbacks));
+  EXPECT_CALL(*conn_pool_callbacks_raw, onFailure()).Times(0);
+  EXPECT_CALL(*conn_pool_callbacks_raw, onSuccess(_)).Times(0);
+  Http::ResponseHeaderMapPtr headers{new Http::TestResponseHeaderMapImpl{{":status", "418"}}};
+  upstream_->responseDecoder().decodeHeaders(std::move(headers), false);
+}
+  
```

Comment:
```suggestion

```

Reply:
I apologize if there is any confusion on my part. I noticed in a previous 2 commit, the line of code was removed and then added back again since you requested me to add it.

Reply:
No problem, I meant to remove the redundant space in the line not the line itself

Reply:
Upon reviewing the code on my end, I am unable to identify any redundant spaces. However, when I attempt to commit the changes, it appears that there is an issue with unnecessary spaces.

Reply:
I think it should work now!